internal/builder/docker.go

```diff
@@ -0,0 +1,130 @@
+package builder
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"runtime"
+
+	"github.com/lucor/fyne-cross/internal/volume"
+)
+
+const (
+	// DefaultDockerImage is the remote docker image
+	// name used by default
+	DefaultDockerImage = "lucor/fyne-cross:1.4.0"
+)
+
+type dockerImageProvider interface {
+	DockerImageName() (string, error)
+}
+
+type dockerRemoteImage struct {
+	Name string
+}
+
+func (d *dockerRemoteImage) DockerImageName() (string, error) {
+	return d.Name, nil
+}
+
+type dockerFile struct {
+	Path string
+}
+
+func (d *dockerFile) DockerImageName() (string, error) {
```

Comment:
Adding support for local image is a nice to have but this mean we need to add both formats for the docker image (remote and local). What about supporting only the remote one for now ?
The build of the local image can be handled with an external script.

Reply:
There's no need to add the Dockefile to the repo. I'm using this Dockerfile:

```
FROM lucor/fyne-cross:1.4.0
RUN apt-get update && apt-get -y install libglib2.0-dev libgtk-3-dev
```

This uses your image and just runs another command on top to install the dependencies.

Reply:
Just realised you meant running a local image by its name. I'll add support for that too.

Reply:
The idea was to support only remote. In this way fyne-cross will be only responsible to all task related to app build. What do you think ?

Reply:
Just to be clear, I'm considering 3 sources of images here:

- Image from name pulled from docker registry (the only supported way and hardcoded for now)
- Image from using an existing image in the system (Docker handles this transparently in the same way as the previous case)
- Image from a Dockerfile

I understand that what you are suggesting is dropping the support for providing a Dockerfile to simplify the code, and instead require the user build the image via their build scripts then pass just the image name to fyne-cross. I don't think that simplifies the code that much, and being able to pass just a Dockerfile and let fyne-cross set it up is way more convenient.

Consider that passing interfaces around is way more maintanable than explicit basic type parameters (strings, ints, bools, etc...) down the line, because you can extend the functions handling them without having to change all the signatures.

Reply:
I see that adding support for a Docker file and building the image inside fyne-cross could be useful.

But this would add an extra feature that we need to maintain or support. I'm thinking of all the problem could be related to building an image itself and not due to the `docker build -f /path/Dockerfile` command :-)

The feature can be addressed with an external step and as you correctly pointed before in any case most of the time you will need to call target specific tools anyway.